src/components/button/Button.base.tsx

```diff
@@ -43,34 +39,19 @@ const ButtonBase: React.FC<PButton> = ({
     }
 
     return (
-        <Shape
-            component={'button'}
-            borderRadius={4}
-            width={width === 'full' ? '100%' : width}
-            height={height}
-            {...rest}
-        >
-            <Grid
-                row
-                component={'span'}
-                alignment={'center'}
-                justify={'center'}
-                padding={Spacing.symmetric(spacing)}
-                flex={1}
+        <button onClick={onClick} className={className}>
+            {icon && iconPosition === 'start' ? <Icon glyph={icon} size={iconSize} /> : null}
```

Comment:
instead of adding the iconPosition prop I would use the leadingIcon/TrailingIcon approach

Reply:
We already discussed this with Dean and decided that, to prevent having icons on both sides, we would rather have 2 separate props to define iconPosition and the actual icon